### ISSUE TITLE
[Flutter] - Crash app on Android when get `permissionSubscriptionState` - Failed to handle method call NullPointerException: Attempt to invoke virtual method

### ISSUE THREAD

**Issue by mrgoonie** [OneSignal/OneSignal-Flutter-SDK#50]
I got this error, not sure how to fix it:

```
I/flutter (18349): [PUSH NOTIFICATION] Getting permissionSubscriptionState
E/MethodChannel#OneSignal(18349): Failed to handle method call
E/MethodChannel#OneSignal(18349): java.lang.NullPointerException: Attempt to invoke virtual method 'com.onesignal.OSSubscriptionState com.onesignal.OSPermissionSubscriptionState.getSubscriptionStatus()' on a null object reference
E/MethodChannel#OneSignal(18349): 	at com.onesignal.flutter.OneSignalSerializer.convertPermissionSubscriptionStateToMap(OneSignalSerializer.java:47)
E/MethodChannel#OneSignal(18349): 	at com.onesignal.flutter.OneSignalPlugin.getPermissionSubscriptionState(OneSignalPlugin.java:167)
E/MethodChannel#OneSignal(18349): 	at com.onesignal.flutter.OneSignalPlugin.onMethodCall(OneSignalPlugin.java:76)
E/MethodChannel#OneSignal(18349): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:200)
E/MethodChannel#OneSignal(18349): 	at io.flutter.view.FlutterNativeView.handlePlatformMessage(FlutterNativeView.java:163)
E/MethodChannel#OneSignal(18349): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#OneSignal(18349): 	at android.os.MessageQueue.next(MessageQueue.java:325)
E/MethodChannel#OneSignal(18349): 	at android.os.Looper.loop(Looper.java:142)
E/MethodChannel#OneSignal(18349): 	at android.app.ActivityThread.main(ActivityThread.java:6494)
E/MethodChannel#OneSignal(18349): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#OneSignal(18349): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
E/MethodChannel#OneSignal(18349): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

Please check!

**Comment by mrgoonie** [comment]
The code was copied from your `/example`:

![image](https://user-images.githubusercontent.com/6857382/51425062-9022e480-1c09-11e9-870e-131c1287eea9.png)

It works just fine on iOS, but when I try it on Android Emulator (Pixel 2), I got that error.

**Comment by mrgoonie** [comment]
Here is the full log:

```
E/flutter (30626): PlatformException(error, Attempt to invoke virtual method 'com.onesignal.OSSubscriptionState com.onesignal.OSPermissionSubscriptionState.getSubscriptionStatus()' on a null object reference, null)
E/flutter (30626): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:551:7)
E/flutter (30626): #1      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:292:18)
E/flutter (30626): <asynchronous suspension>
E/flutter (30626): #2      OneSignal.getPermissionSubscriptionState (package:onesignal/onesignal.dart:209:24)
E/flutter (30626): <asynchronous suspension>
E/flutter (30626): #3      PushNotification.subscribe (package:begreater/config/push_notification.dart:95:22)
E/flutter (30626): <asynchronous suspension>
E/flutter (30626): #4      new Future.delayed.<anonymous closure> (dart:async/future.dart:316:39)
E/flutter (30626): #5      Timer._createTimer.<anonymous closure> (dart:async/runtime/libtimer_patch.dart:21:15)
E/flutter (30626): #6      _Timer._runTimers (dart:isolate/runtime/libtimer_impl.dart:382:19)
E/flutter (30626): #7      _Timer._handleMessage (dart:isolate/runtime/libtimer_impl.dart:416:5)
E/flutter (30626): #8      _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
```

**Comment by mrgoonie** [comment]
Here is the full log:

```
E/OneSignal(30626): OneSignal.init has not been called. Could not get OSPermissionSubscriptionState
E/MethodChannel#OneSignal(30626): Failed to handle method call
E/MethodChannel#OneSignal(30626): java.lang.NullPointerException: Attempt to invoke virtual method 'com.onesignal.OSSubscriptionState com.onesignal.OSPermissionSubscriptionState.getSubscriptionStatus()' on a null object reference
E/MethodChannel#OneSignal(30626): 	at com.onesignal.flutter.OneSignalSerializer.convertPermissionSubscriptionStateToMap(OneSignalSerializer.java:47)
E/MethodChannel#OneSignal(30626): 	at com.onesignal.flutter.OneSignalPlugin.getPermissionSubscriptionState(OneSignalPlugin.java:167)
E/MethodChannel#OneSignal(30626): 	at com.onesignal.flutter.OneSignalPlugin.onMethodCall(OneSignalPlugin.java:76)
E/MethodChannel#OneSignal(30626): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:200)
E/MethodChannel#OneSignal(30626): 	at io.flutter.view.FlutterNativeView.handlePlatformMessage(FlutterNativeView.java:163)
E/MethodChannel#OneSignal(30626): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#OneSignal(30626): 	at android.os.MessageQueue.next(MessageQueue.java:325)
E/MethodChannel#OneSignal(30626): 	at android.os.Looper.loop(Looper.java:142)
E/MethodChannel#OneSignal(30626): 	at android.app.ActivityThread.main(ActivityThread.java:6494)
E/MethodChannel#OneSignal(30626): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#OneSignal(30626): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
E/MethodChannel#OneSignal(30626): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
E/flutter (30626): [ERROR:flutter/shell/common/shell.cc(184)] Dart Error: Unhandled exception:
E/flutter (30626): PlatformException(error, Attempt to invoke virtual method 'com.onesignal.OSSubscriptionState com.onesignal.OSPermissionSubscriptionState.getSubscriptionStatus()' on a null object reference, null)
E/flutter (30626): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:551:7)
E/flutter (30626): #1      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:292:18)
E/flutter (30626): <asynchronous suspension>
E/flutter (30626): #2      OneSignal.getPermissionSubscriptionState (package:onesignal/onesignal.dart:209:24)
E/flutter (30626): <asynchronous suspension>
E/flutter (30626): #3      PushNotification.subscribe (package:begreater/config/push_notification.dart:95:22)
E/flutter (30626): <asynchronous suspension>
E/flutter (30626): #4      new Future.delayed.<anonymous closure> (dart:async/future.dart:316:39)
E/flutter (30626): #5      Timer._createTimer.<anonymous closure> (dart:async/runtime/libtimer_patch.dart:21:15)
W/OneSignal(30626): 	at android.app.ActivityThread.main(ActivityThread.java:6494)
W/OneSignal(30626): 	at java.lang.reflect.Method.invoke(Native Method)
W/OneSignal(30626): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
W/OneSignal(30626): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
I/flutter (30626): [PUSH NOTIFICATION] Getting permissionSubscriptionState
E/flutter (30626): #6      _Timer._runTimers (dart:isolate/runtime/libtimer_impl.dart:382:19)
E/flutter (30626): #7      _Timer._handleMessage (dart:isolate/runtime/libtimer_impl.dart:416:5)
E/flutter (30626): #8      _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
```

**Comment by rgomezp** [comment]
Hello, 
It looks like the method cannot be called because the `OSPermissionSubscriptionState` is null. Where are you initializing OneSignal? It would be helpful if you could include your entire dart file.

**Comment by mrgoonie** [comment]
@rgomezp I can't post the dart file because my app structure is complex, but I put the initial code of OneSignal in initState of the first screen. I tried to put it into a button pressed event as well, same issue. Can it be somehow related to Android SDK version? (mine is 27)

**Comment by rgomezp** [comment]
@mrgoonie we are still looking into the issue and will fix it in the next release. thanks for bringing it to our attention!

**Comment by ashkryab** [comment]
call  OneSignal.shared.getPermissionSubscriptionState()  after 

await OneSignal.shared.init(currentFlavour.oneSignalKey,
        iOSSettings: <OSiOSSettings, dynamic>{
          OSiOSSettings.autoPrompt: true,
          OSiOSSettings.inAppLaunchUrl: true
        });

**Comment by rgomezp** [comment]
Howdy y'all,
This is quite an old issue so will close. Thanks for your comment @ashkryab

**Comment by rafaelviny** [comment]
@mrgoonie  
I am experiencing the same problem, did you manage to solve it?

**Comment by fede1295** [comment]
OneSignal.shared.setRequiresUserPrivacyConsent(false);

this solved the problem for me

**Comment by tamoyal** [comment]
I get a similar issue:
```
PlatformException(error, Attempt to invoke virtual method 'boolean java.lang.Object.equals(java.lang.Object)' on a null object reference, null, java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Object.equals(java.lang.Object)' on a null object reference
	at com.onesignal.OSObservable.d(OSObservable.java:3)
	at com.onesignal.OneSignal.removeSubscriptionObserver(OneSignal.java:3)
	at com.onesignal.flutter.OneSignalPlugin.addObservers(OneSignalPlugin.java:1)
	at com.onesignal.flutter.OneSignalPlugin.setAppId(OneSignalPlugin.java:8)
	at com.onesignal.flutter.OneSignalPlugin.onMethodCall(OneSignalPlugin.java:2)
	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:2)
	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:5)
	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:2)
	at android.os.MessageQueue.nativePollOnce(Native Method)
	at android.os.MessageQueue.next(MessageQueue.java:323)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5421)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
)
```

**Comment by rodolfosaraiva** [comment]
same issue of @tamoyal , did you find a solution?

**Comment by nan-li** [comment]
Hi @tamoyal @rodolfosaraiva @cntoss,

Sorry for the delayed response. I see the new reports are for `removeObserver` called by `removeSubscriptionObserver`. This was [also reported](https://github.com/OneSignal/OneSignal-Android-SDK/issues/1524) in the OneSignal-Android-SDK.

The fix for that exception has been released in `OneSignal-Flutter-SDK` [version 3.2.8](https://github.com/OneSignal/OneSignal-Flutter-SDK/releases/tag/3.2.8).

Since there has not been activity on the original issue about `getSubscriptionStatus` in over a year and in that time, the SDK has moved from 2.x.x to 3.x.x, I will close this issue.
